chaincode/src/oracle/saveOracleDefinition.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) Gala Games Inc. All rights reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { ChainError, ErrorCode, OracleDefinition, UnauthorizedError } from "@gala-chain/api";
+import { plainToInstance } from "class-transformer";
+
+import { GalaChainContext } from "../types";
+import { getObjectByKey, putChainObject } from "../utils";
+
+export interface IOracleDefinition {
+  name: string;
+  authorities: string[];
+}
+
+export async function saveOracleDefinition(
+  ctx: GalaChainContext,
+  data: IOracleDefinition
+): Promise<OracleDefinition> {
+  const existingOracle = await getObjectByKey(
+    ctx,
+    OracleDefinition,
+    OracleDefinition.getCompositeKeyFromParts(OracleDefinition.INDEX_KEY, [data.name])
+  ).catch((e) => {
+    const chainError = ChainError.from(e);
+    if (chainError.matches(ErrorCode.NOT_FOUND)) {
+      return undefined;
+    } else {
+      throw chainError;
+    }
```

Comment:
I believe this is a common pattern we have in our code, I wonder how we can improve our `ChainError` to address that. Maybe something like this would be useful?

```
  ).catch((e) => ChainError.ignore(e, ErrorCode.NOT_FOUND));
```

I think we should create an issue for that, or maybe some common other utils